conans/client/tools/scm.py

```diff
@@ -146,4 +171,99 @@ def get_branch(self):
             branch = status.splitlines()[0].split("...")[0].strip("#").strip()
             return branch
         except Exception as e:
-            raise ConanException("Unable to get git branch from %s\n%s" % (self.folder, str(e)))
+            raise ConanException("Unable to get git branch from %s: %s" % (self.folder, str(e)))
+
+
+class SVN(SCMBase):
+    cmd_command = "svn"
+
+    def __init__(self, folder=None, runner=None, *args, **kwargs):
+        def runner_no_strip(command):
+            return decode_text(subprocess.check_output(command, shell=True))
+        runner = runner or runner_no_strip
+        super(SVN, self).__init__(folder=folder, runner=runner, *args, **kwargs)
+
+    def run(self, command):
+        # Ensure we always pass some params
+        return super(SVN, self).run(command="{} --no-auth-cache --non-interactive".format(command))
+
+    def clone(self, url, submodule=None):
+        assert submodule is None, "Argument not handled"
+        assert os.path.exists(self.folder), "It guaranteed to exists according to SCMBase::__init__"
+        params = " --trust-server-cert-failures=unknown-ca" if not self._verify_ssl else ""
+
+        output = ""
+        if not os.path.exists(os.path.join(self.folder, '.svn')):
+            url = self.get_url_with_credentials(url)
+            command = 'co "{url}" . {params}'.format(url=url, params=params)
+            output += self.run(command)
+        output += self.run("revert . --recursive {params}".format(params=params))
+        # output += self.run("cleanup . --remove-unversioned --remove-ignored {params}".format(params=params))
+
+        return output
+
+    def checkout(self, element, submodule=None):
+        # Element can only be a revision number
+        return self.run("update -r {rev}".format(rev=element))
```

Comment:
Naming clash (same as for "clone" which is not an SVN command).
From an SVN point of view a checkout is the same an git clone & checkout. 
Here it is "just" svn update which is nearly the same as git pull. A git checkout is normally used to switch branches.

So probably it would make sense to check the common SCM API and find method names which are more SCM-system-used unspecific.
But I know it is hard to find a good compromise and method naming for both, SVN and Git.

Reply:
Yes, I agree with you. At this moment the naming follows Git conventions and the workflow in Conan reflects that of Git too; SVN has arrived later and has made an effort to fit into it. At this moment I think that we need to keep it as it is for backward compatibility, but at some point in time it would be useful to write down the use cases and refactor some classes. Some ideas:
- `Git` and `SVN` classes do not need to have the same interface. Each one should we a wrapper over the cli tool and we provided as standalone Conan tools (`from conans.tools import SVN, Git`).
- `SCM` needs its own set of methods that makes sense for the feature workflow, not necessarely related tot SVN or Git
- Write adaptors `Git2SCM` and `SVN2SCM` to use each _tool_ for the SCM feature.

Maybe this should be reported as an [engineering issue](https://github.com/conan-io/conan/issues?q=is%3Aissue+is%3Aopen+label%3Aengineering).

Reply:
> At this moment I think that we need to keep it as it is for backward compatibility,

Remember that this feature is experimental. If any change is really needed we can introduce it.

- First point, True. Change it in the current PR.
- Second point: True, but the usage of the `scm` class is mostly internal in Conan. We could do any change but I think the interface of the scm is good enough for us.
- I think it can be a little overkilling. The SCM class is tiny, I would just "if svn" "if git" and call the correct interface. The most important (less breaking) is keeping (point 1) the correct interface in `svn` class, future refactors could be done without breaking anything.

Reply:
I partially agree. But Git and SVN needs same API provided by SCM because of base recipes using python_requires.
@lasote what do you think?

Reply:
Sure. The `SVN` class will have its own interface, the `Git` class will have its own interface, the `SCM` class will manage with a simple "if" which method of the `Git/SVN` call.